app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -521,16 +526,11 @@ private void initPlayer(final boolean playOnReady) {
         // Setup UIs
         UIs.call(PlayerUi::initPlayer);
 
-        // enable media tunneling
-        if (DEBUG && PreferenceManager.getDefaultSharedPreferences(context)
+        // Disable media tunneling if requested by the user from ExoPlayer settings
+        if (!PreferenceManager.getDefaultSharedPreferences(context)
                 .getBoolean(context.getString(R.string.disable_media_tunneling_key), false)) {
```

Comment:
So I would make sure the preference does not get auto-created, and then use `!DeviceUtils.shouldSupportMediaTunneling()` instead of `false` as the value to use when the preference is not set.